Build failure: libuv

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

Building libuv from inside of my act-runner (gitea actions) causes the libuv test for thread affinity to fail an assertion: `# Assertion failed in test/test-thread-affinity.c on line 73: cpumask[0] && "test must be run with cpu 0 affinity"`

This package is building on my gitea runner, which is a lxc container that does modify affinity. For example,

```
[root@git-runner1:~]# taskset -cp $$
pid 4030471's current affinity list: 5,7
```


### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
[root@git-runner1:~]# nix log /nix/store/34pv1kp7z7mqhx8q9ri9i6x214k9bn1m-libuv-1.49.2.drv

Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/i5ks3w8xjbwjmz7gc1cvakz3c4nrkn4v-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
/nix/store/skdaq4fi7y0ipp9ij1fa3iznz3jbp399-gnused-4.9/bin/sed: couldn't flush stdout: Broken pipe
+ libtoolize --copy --force
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
+ aclocal -I m4
+ autoconf
configure.ac:45: warning: The macro 'AC_PROG_LIBTOOL' is obsolete.
configure.ac:45: You should run autoupdate.
m4/libtool.m4:100: AC_PROG_LIBTOOL is expanded from...
configure.ac:45: the top level
configure.ac:48: warning: $as_echo is obsolete; use AS_ECHO(["message"]) instead
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
m4/ax_pthread.m4:88: AX_PTHREAD is expanded from...
configure.ac:48: the top level
+ automake --add-missing --copy
configure.ac:43: installing './ar-lib'
configure.ac:26: installing './compile'
configure.ac:23: installing './config.guess'
configure.ac:23: installing './config.sub'
configure.ac:21: installing './install-sh'
configure.ac:21: installing './missing'
Makefile.am: installing './depcomp'
fixing libtool script ./ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/lbyr47xikzcaz9zny4dnjaqnxigimcq2-bash-5.2p37/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/1nckbwk9yck3bb0njmmn7w08k9qpvj8v-libuv-1.49.2 --bindir=/nix/store/1nckbwk9yck3bb0njmmn7w08k9qpvj8v-libuv-1.49.2/bin --sbindir=/nix/store/1nckbwk9yck3bb0njmmn7w08k9qpvj8v-libuv-1.49.2/sbin --includedir=/nix/store/6z374d704fz9jjqr8yjfrl663kq96kmx-libuv-1.49.2-dev/include --oldincludedir=/nix/store/6z374d704fz9jjqr8yjfrl663kq96kmx-libuv-1.49.2-dev/include --mandir=/nix/store/1nckbwk9yck3bb0njmmn7w08k9qpvj8v-libuv-1.49.2/share/man --infodir=/nix/store/1nckbwk9yck3bb0njmmn7w08k9qpvj8v-libuv-1.49.2/share/info --docdir=/nix/store/1nckbwk9yck3bb0njmmn7w08k9qpvj8v-libuv-1.49.2/share/doc/libuv --libdir=/nix/store/1nckbwk9yck3bb0njmmn7w08k9qpvj8v-libuv-1.49.2/lib --libexecdir=/nix/store/1nckbwk9yck3bb0njmmn7w08k9qpvj8v-libuv-1.49.2/libexec --localedir=/nix/store/1nckbwk9yck3bb0njmmn7w08k9qpvj8v-libuv-1.49.2/share/locale
checking for a BSD-compatible install... /nix/store/35wf5ssjrfvllcsdhr1n5i471c1pd0zs-coreutils-9.5/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/35wf5ssjrfvllcsdhr1n5i471c1pd0zs-coreutils-9.5/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... none
checking for gcc way to treat warnings as errors... -Werror
checking if gcc supports __attribute__(( visibility("default") ))... yes
checking if gcc supports -fvisibility=hidden... yes
checking if gcc supports -fno-strict-aliasing flag... yes
checking if gcc supports -fno-strict-aliasing flag... yes
checking if gcc supports -g flag... yes
checking if gcc supports -std=gnu89 flag... yes
checking if gcc supports -Wall flag... yes
checking if gcc supports -Wextra flag... yes
checking if gcc supports -Wno-long-long flag... yes
checking if gcc supports -Wno-unused-parameter flag... yes
checking if gcc supports -Wstrict-prototypes flag... yes
checking the archiver (ar) interface... ar
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/skdaq4fi7y0ipp9ij1fa3iznz3jbp399-gnused-4.9/bin/sed
checking for grep that handles long lines and -e... /nix/store/228r34facard7h1a358mpzal77y7znvz-gnugrep-3.11/bin/grep
checking for egrep... /nix/store/228r34facard7h1a358mpzal77y7znvz-gnugrep-3.11/bin/grep -E
checking for fgrep... /nix/store/228r34facard7h1a358mpzal77y7znvz-gnugrep-3.11/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/35wf5ssjrfvllcsdhr1n5i471c1pd0zs-coreutils-9.5/bin/dd
checking how to truncate binary pipes... /nix/store/35wf5ssjrfvllcsdhr1n5i471c1pd0zs-coreutils-9.5/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking whether make supports nested variables... (cached) yes
checking how to run the C preprocessor... gcc -E
checking for egrep -e... (cached) /nix/store/228r34facard7h1a358mpzal77y7znvz-gnugrep-3.11/bin/grep -E
checking whether gcc is Clang... no
checking whether pthreads work with -pthread... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking for library containing dlopen... none required
checking for library containing kstat_lookup... no
checking for library containing gethostbyname... none required
checking for library containing perfstat_cpu... no
checking for library containing clock_gettime... none required
checking for library containing sendfile... none required
checking for library containing socket... none required
checking for gcc option to enable large file support... none needed
checking for sys/ahafs_evProds.h... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libuv.pc
config.status: executing depfiles commands
config.status: executing libtool commands
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j2 SHELL=/nix/store/lbyr47xikzcaz9zny4dnjaqnxigimcq2-bash-5.2p37/bin/bash
  CC       src/libuv_la-fs-poll.lo
  CC       src/libuv_la-idna.lo
  CC       src/libuv_la-inet.lo
  CC       src/libuv_la-random.lo
  CC       src/libuv_la-strscpy.lo
  CC       src/libuv_la-thread-common.lo
  CC       src/libuv_la-threadpool.lo
  CC       src/libuv_la-timer.lo
  CC       src/libuv_la-uv-data-getter-setters.lo
  CC       src/libuv_la-uv-common.lo
  CC       src/libuv_la-version.lo
  CC       src/libuv_la-strtok.lo
  CC       src/unix/libuv_la-async.lo
  CC       src/unix/libuv_la-core.lo
  CC       src/unix/libuv_la-dl.lo
  CC       src/unix/libuv_la-fs.lo
  CC       src/unix/libuv_la-getaddrinfo.lo
  CC       src/unix/libuv_la-getnameinfo.lo
  CC       src/unix/libuv_la-loop-watcher.lo
  CC       src/unix/libuv_la-loop.lo
  CC       src/unix/libuv_la-pipe.lo
  CC       src/unix/libuv_la-poll.lo
  CC       src/unix/libuv_la-process.lo
  CC       src/unix/libuv_la-random-devurandom.lo
  CC       src/unix/libuv_la-signal.lo
  CC       src/unix/libuv_la-stream.lo
  CC       src/unix/libuv_la-tcp.lo
  CC       src/unix/libuv_la-thread.lo
  CC       src/unix/libuv_la-tty.lo
  CC       src/unix/libuv_la-udp.lo
  CC       src/unix/libuv_la-linux.lo
  CC       src/unix/libuv_la-procfs-exepath.lo
  CC       src/unix/libuv_la-proctitle.lo
  CC       src/unix/libuv_la-random-getrandom.lo
  CC       src/unix/libuv_la-random-sysctl-linux.lo
  CCLD     libuv.la
Running phase: checkPhase
@nix { "action": "setPhase", "phase": "checkPhase" }
check flags: -j2 SHELL=/nix/store/lbyr47xikzcaz9zny4dnjaqnxigimcq2-bash-5.2p37/bin/bash VERBOSE=y check
make  test/run-tests
make[1]: Entering directory '/build/source'
  CC       test/run_tests-blackhole-server.o
  CC       test/run_tests-echo-server.o
  CC       test/run_tests-run-tests.o
  CC       test/run_tests-runner.o
  CC       test/run_tests-test-active.o
  CC       test/run_tests-test-async.o
  CC       test/run_tests-test-async-null-cb.o
  CC       test/run_tests-test-barrier.o
  CC       test/run_tests-test-callback-stack.o
  CC       test/run_tests-test-close-fd.o
  CC       test/run_tests-test-close-order.o
  CC       test/run_tests-test-condvar.o
  CC       test/run_tests-test-connect-unspecified.o
  CC       test/run_tests-test-connection-fail.o
  CC       test/run_tests-test-cwd-and-chdir.o
  CC       test/run_tests-test-default-loop-close.o
  CC       test/run_tests-test-delayed-accept.o
  CC       test/run_tests-test-dlerror.o
  CC       test/run_tests-test-eintr-handling.o
  CC       test/run_tests-test-embed.o
  CC       test/run_tests-test-emfile.o
  CC       test/run_tests-test-env-vars.o
  CC       test/run_tests-test-error.o
  CC       test/run_tests-test-fail-always.o
  CC       test/run_tests-test-fs-copyfile.o
  CC       test/run_tests-test-fs-event.o
  CC       test/run_tests-test-fs-poll.o
  CC       test/run_tests-test-fs.o
  CC       test/run_tests-test-fs-readdir.o
  CC       test/run_tests-test-fs-fd-hash.o
  CC       test/run_tests-test-fs-open-flags.o
  CC       test/run_tests-test-fork.o
  CC       test/run_tests-test-getters-setters.o
  CC       test/run_tests-test-get-currentexe.o
  CC       test/run_tests-test-get-loadavg.o
  CC       test/run_tests-test-get-memory.o
  CC       test/run_tests-test-get-passwd.o
  CC       test/run_tests-test-getaddrinfo.o
  CC       test/run_tests-test-gethostname.o
  CC       test/run_tests-test-getnameinfo.o
  CC       test/run_tests-test-getsockname.o
  CC       test/run_tests-test-gettimeofday.o
  CC       test/run_tests-test-handle-fileno.o
  CC       test/run_tests-test-homedir.o
  CC       test/run_tests-test-hrtime.o
  CC       test/run_tests-test-idle.o
  CC       test/run_tests-test-idna.o
  CC       test/run_tests-test-iouring-pollhup.o
  CC       test/run_tests-test-ip4-addr.o
  CC       test/run_tests-test-ip6-addr.o
  CC       test/run_tests-test-ip-name.o
  CC       test/run_tests-test-ipc-heavy-traffic-deadlock-bug.o
  CC       test/run_tests-test-ipc-send-recv.o
  CC       test/run_tests-test-ipc.o
  CC       test/run_tests-test-loop-handles.o
  CC       test/run_tests-test-loop-alive.o
  CC       test/run_tests-test-loop-close.o
  CC       test/run_tests-test-loop-stop.o
  CC       test/run_tests-test-loop-time.o
  CC       test/run_tests-test-loop-configure.o
  CC       test/run_tests-test-metrics.o
  CC       test/run_tests-test-multiple-listen.o
  CC       test/run_tests-test-mutexes.o
  CC       test/run_tests-test-not-readable-nor-writable-on-read-error.o
  CC       test/run_tests-test-not-writable-after-shutdown.o
  CC       test/run_tests-test-osx-select.o
  CC       test/run_tests-test-pass-always.o
  CC       test/run_tests-test-ping-pong.o
  CC       test/run_tests-test-pipe-bind-error.o
  CC       test/run_tests-test-pipe-connect-error.o
  CC       test/run_tests-test-pipe-connect-multiple.o
  CC       test/run_tests-test-pipe-connect-prepare.o
  CC       test/run_tests-test-pipe-getsockname.o
  CC       test/run_tests-test-pipe-pending-instances.o
  CC       test/run_tests-test-pipe-sendmsg.o
  CC       test/run_tests-test-pipe-server-close.o
  CC       test/run_tests-test-pipe-close-stdout-read-stdin.o
  CC       test/run_tests-test-pipe-set-non-blocking.o
  CC       test/run_tests-test-pipe-set-fchmod.o
  CC       test/run_tests-test-platform-output.o
  CC       test/run_tests-test-poll.o
  CC       test/run_tests-test-poll-close.o
  CC       test/run_tests-test-poll-close-doesnt-corrupt-stack.o
  CC       test/run_tests-test-poll-closesocket.o
  CC       test/run_tests-test-poll-multiple-handles.o
  CC       test/run_tests-test-poll-oob.o
  CC       test/run_tests-test-process-priority.o
  CC       test/run_tests-test-process-title.o
  CC       test/run_tests-test-process-title-threadsafe.o
  CC       test/run_tests-test-queue-foreach-delete.o
  CC       test/run_tests-test-random.o
  CC       test/run_tests-test-readable-on-eof.o
  CC       test/run_tests-test-ref.o
  CC       test/run_tests-test-run-nowait.o
  CC       test/run_tests-test-run-once.o
  CC       test/run_tests-test-semaphore.o
  CC       test/run_tests-test-shutdown-close.o
  CC       test/run_tests-test-shutdown-eof.o
  CC       test/run_tests-test-shutdown-simultaneous.o
  CC       test/run_tests-test-shutdown-twice.o
  CC       test/run_tests-test-signal-multiple-loops.o
  CC       test/run_tests-test-signal-pending-on-close.o
  CC       test/run_tests-test-signal.o
  CC       test/run_tests-test-socket-buffer-size.o
  CC       test/run_tests-test-spawn.o
  CC       test/run_tests-test-stdio-over-pipes.o
  CC       test/run_tests-test-strscpy.o
  CC       test/run_tests-test-strtok.o
  CC       test/run_tests-test-tcp-alloc-cb-fail.o
  CC       test/run_tests-test-tcp-bind-error.o
  CC       test/run_tests-test-tcp-bind6-error.o
  CC       test/run_tests-test-tcp-close-accept.o
  CC       test/run_tests-test-tcp-close-while-connecting.o
  CC       test/run_tests-test-tcp-close-after-read-timeout.o
  CC       test/run_tests-test-tcp-close.o
  CC       test/run_tests-test-tcp-close-reset.o
  CC       test/run_tests-test-tcp-create-socket-early.o
  CC       test/run_tests-test-tcp-connect-error-after-write.o
  CC       test/run_tests-test-tcp-connect-error.o
  CC       test/run_tests-test-tcp-connect-timeout.o
  CC       test/run_tests-test-tcp-connect6-error.o
  CC       test/run_tests-test-tcp-flags.o
  CC       test/run_tests-test-tcp-open.o
  CC       test/run_tests-test-tcp-read-stop.o
  CC       test/run_tests-test-tcp-reuseport.o
  CC       test/run_tests-test-tcp-read-stop-start.o
  CC       test/run_tests-test-tcp-rst.o
  CC       test/run_tests-test-tcp-shutdown-after-write.o
  CC       test/run_tests-test-tcp-unexpected-read.o
  CC       test/run_tests-test-tcp-oob.o
  CC       test/run_tests-test-tcp-write-to-half-open-connection.o
  CC       test/run_tests-test-tcp-write-after-connect.o
  CC       test/run_tests-test-tcp-writealot.o
  CC       test/run_tests-test-tcp-write-fail.o
  CC       test/run_tests-test-tcp-try-write.o
  CC       test/run_tests-test-tcp-write-in-a-row.o
  CC       test/run_tests-test-tcp-try-write-error.o
  CC       test/run_tests-test-tcp-write-queue-order.o
  CC       test/run_tests-test-test-macros.o
  CC       test/run_tests-test-thread-equal.o
  CC       test/run_tests-test-thread.o
  CC       test/run_tests-test-thread-affinity.o
  CC       test/run_tests-test-thread-priority.o
  CC       test/run_tests-test-threadpool-cancel.o
  CC       test/run_tests-test-threadpool.o
  CC       test/run_tests-test-timer-again.o
  CC       test/run_tests-test-timer-from-check.o
  CC       test/run_tests-test-timer.o
  CC       test/run_tests-test-tmpdir.o
  CC       test/run_tests-test-tty-duplicate-key.o
  CC       test/run_tests-test-tty-escape-sequence-processing.o
  CC       test/run_tests-test-tty.o
  CC       test/run_tests-test-udp-alloc-cb-fail.o
  CC       test/run_tests-test-udp-bind.o
  CC       test/run_tests-test-udp-connect.o
  CC       test/run_tests-test-udp-connect6.o
  CC       test/run_tests-test-udp-create-socket-early.o
  CC       test/run_tests-test-udp-dgram-too-big.o
  CC       test/run_tests-test-udp-ipv6.o
  CC       test/run_tests-test-udp-mmsg.o
  CC       test/run_tests-test-udp-multicast-interface.o
  CC       test/run_tests-test-udp-multicast-interface6.o
  CC       test/run_tests-test-udp-multicast-join.o
  CC       test/run_tests-test-udp-multicast-join6.o
  CC       test/run_tests-test-udp-multicast-ttl.o
  CC       test/run_tests-test-udp-open.o
  CC       test/run_tests-test-udp-options.o
  CC       test/run_tests-test-udp-send-and-recv.o
  CC       test/run_tests-test-udp-send-hang-loop.o
  CC       test/run_tests-test-udp-send-immediate.o
  CC       test/run_tests-test-udp-sendmmsg-error.o
  CC       test/run_tests-test-udp-send-unreachable.o
  CC       test/run_tests-test-udp-try-send.o
  CC       test/run_tests-test-udp-recv-in-a-row.o
  CC       test/run_tests-test-udp-reuseport.o
  CC       test/run_tests-test-uname.o
  CC       test/run_tests-test-walk-handles.o
  CC       test/run_tests-test-watcher-cross-stop.o
  CC       test/run_tests-runner-unix.o
  CCLD     test/run-tests
make[1]: Leaving directory '/build/source'
make  check-TESTS
make[1]: Entering directory '/build/source'
1..417
ok 1 - platform_output
# Output from process `platform_output`:
# uv_get_process_title: /build/source/test/.libs/lt-run-tests
# uv_cwd: /build/source
# uv_resident_set_memory: 1572864
# uv_uptime: 1198233.060000
# uv_getrusage:
#   user: 0 sec 499 microsec
#   system: 0 sec 0 microsec
#   page faults: 4
#   maximum resident set size: 1536
# uv_available_parallelism: 2
# uv_cpu_info:
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#   speed: 0
#   times.sys: 16355240
#   times.user: 45293150
#   times.idle: 1123208620
#   times.irq: 3029730
#   times.nice: 595370
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#   speed: 0
#   times.sys: 15757390
#   times.user: 61131750
#   times.idle: 1111939860
#   times.irq: 2563860
#   times.nice: 575960
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#   speed: 0
#   times.sys: 19810640
#   times.user: 42427600
#   times.idle: 1128714300
#   times.irq: 3282980
#   times.nice: 676010
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#   speed: 0
#   times.sys: 13747070
#   times.user: 64166910
#   times.idle: 1112322230
#   times.irq: 2137580
#   times.nice: 1212260
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#  
#  speed: 0
#   times.sys: 18987860
#   times.user: 24846300
#   times.idle: 1145214530
#   times.irq: 3666070
#   times.nice: 652850
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#   speed: 0
#   times.sys: 13881540
#   times.user: 20617990
#   times.idle: 1154572610
#   times.irq: 2217840
#   times.nice: 1207940
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#   speed: 0
#   times.sys: 18933850
#   times.user: 17397640
#   times.idle: 1152770120
#   times.irq: 3056070
#   times.nice: 610710
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#   speed: 0
#   times.sys: 16353460
#   times.user: 27361060
#   times.idle: 1146480560
#   times.irq: 2826510
#   times.nice: 18240
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#   speed: 0
#   times.sys: 20524190
#   times.user: 15417860
#   times.idle: 1154629750
#   times.irq: 3169330
#   times.nice: 1122260
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#   speed: 0
#   times.sys: 15968890
#   times.user: 25554350
#   times.idle: 1150321130
#   times.irq: 2238900
#   times.nice: 14260
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#   speed: 0
#   times.sys: 19652370
#   times.user: 17564690
#   times.idle: 1148740020
#   times.irq: 4417460
#   times.nice: 16370
#   model: 12th Gen Intel(R) Core(TM) i5-12600K
#   speed: 0
#   times.sys: 12970080
#   times.user: 15213370
#   times.idle: 1162832970
#   times.irq: 2086640
#   times.nice: 594060
# uv_interface_addresses:
#   name: lo
#   internal: 1
#   physical address: 00:00:00:00:00:00
#   address: 127.0.0.1
#   netmask: 255.0.0.0
#   name: lo
#   internal: 1
#   physical address: 00:00:00:00:00:00
#   address: ::1
#   netmask: ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff
# uv_os_get_passwd:
#   euid: 1000
#   gid: 100 (nixbld)
#     members: [ ]
#   username: nixbld
#   shell: /noshell
#   home directory: /build
# uv_os_getpid: 4578
# uv_os_getppid: 4552
# uv_os_uname:
#   sysname: Linux
#   release: 6.6.63
#   version: #1-NixOS SMP PREEMPT_DYNAMIC Fri Nov 22 14:38:37 UTC 2024
#   machine: x86_64
ok 2 - active
ok 3 - async
ok 4 - async_null_cb
ok 5 - async_ref
ok 6 - barrier_1
ok 7 - barrier_2
ok 8 - barrier_3
ok 9 - barrier_serial_thread
ok 10 - barrier_serial_thread_single
ok 11 - callback_stack
ok 12 - check_ref
ok 13 - clock_gettime
ok 14 - close_fd
ok 15 - close_order
ok 16 - closed_fd_events
ok 17 - condvar_1
ok 18 - condvar_2
ok 19 - condvar_3
ok 20 - condvar_4
ok 21 - condvar_5
ok 22 - connect_unspecified
ok 23 - connection_fail
ok 24 - connection_fail_doesnt_auto_close
ok 25 - cwd_and_chdir
ok 26 - default_loop_close
ok 27 - delayed_accept
ok 28 - dlerror
ok 29 - eintr_handling
ok 30 - embed
ok 31 - emfile
ok 32 - env_vars
ok 33 - error_message
ok 34 - fork_close_signal_in_child
ok 35 - fork_fs_events_child
ok 36 - fork_fs_events_child_dir
ok 37 - fork_fs_events_file_parent_child
ok 38 - fork_signal_to_child
ok 39 - fork_signal_to_child_closed
ok 40 - fork_socketpair
ok 41 - fork_socketpair_started
ok 42 - fork_threadpool_queue_work_simple
ok 43 - fork_timer
ok 44 - fs_access
ok 45 - fs_async_dir
ok 46 - fs_async_sendfile
ok 47 - fs_async_sendfile_nodata
ok 48 - fs_chmod
ok 49 - fs_copyfile
ok 50 - fs_event_close_in_callback
ok 51 - fs_event_close_with_pending_delete_event
ok 52 - fs_event_close_with_pending_event
ok 53 - fs_event_getpath
ok 54 - fs_event_immediate_close
ok 55 - fs_event_no_callback_after_close
ok 56 - fs_event_no_callback_on_close
ok 57 - fs_event_ref
ok 58 - fs_event_start_and_close
ok 59 - fs_event_stop_in_cb
ok 60 - fs_event_watch_delete_dir
ok 61 - fs_event_watch_dir
ok 62 - fs_event_watch_dir_recursive # SKIP Recursive directory watching not supported on this platform.
ok 63 - fs_event_watch_file
ok 64 - fs_event_watch_file_current_dir
ok 65 - fs_event_watch_file_exact_path
ok 66 - fs_event_watch_file_twice
ok 67 - fs_event_watch_invalid_path
ok 68 - fs_file_async
ok 69 - fs_file_loop
ok 70 - fs_file_nametoolong
ok 71 - fs_file_noent
ok 72 - fs_file_open_append
ok 73 - fs_file_pos_after_op_with_offset
ok 74 - fs_file_sync
ok 75 - fs_file_write_null_buffer
ok 76 - fs_futime
ok 77 - fs_get_system_error
ok 78 - fs_link
ok 79 - fs_lutime
ok 80 - fs_mkdtemp
ok 81 - fs_mkstemp
ok 82 - fs_null_req
ok 83 - fs_open_dir
ok 84 - fs_partial_read
ok 85 - fs_partial_write
ok 86 - fs_poll
ok 87 - fs_poll_close_request
ok 88 - fs_poll_close_request_multi_start_stop
ok 89 - fs_poll_close_request_multi_stop_start
ok 90 - fs_poll_close_request_stop_when_active
ok 91 - fs_poll_getpath
ok 92 - fs_poll_ref
ok 93 - fs_posix_delete
ok 94 - fs_read_bufs
ok 95 - fs_read_dir
ok 96 - fs_read_file_eof
ok 97 - fs_read_write_null_arguments
ok 98 - fs_readdir_empty_dir
ok 99 - fs_readdir_file
ok 100 - fs_readdir_non_empty_dir
ok 101 - fs_readdir_non_existing_dir
ok 102 - fs_readlink
ok 103 - fs_realpath
ok 104 - fs_rename_to_existing_file
ok 105 - fs_scandir_early_exit
ok 106 - fs_scandir_empty_dir
ok 107 - fs_scandir_file
ok 108 - fs_scandir_non_existent_dir
ok 109 - fs_stat_batch_multiple
ok 110 - fs_stat_missing_path
ok 111 - fs_statfs
ok 112 - fs_symlink
ok 113 - fs_symlink_dir
ok 114 - fs_unlink_readonly
ok 115 - fs_utime
ok 116 - fs_utime_round
ok 117 - fs_write_alotof_bufs
ok 118 - fs_write_alotof_bufs_with_offset
ok 119 - fs_write_multiple_bufs
ok 120 - get_currentexe
ok 121 - get_group
ok 122 - get_loadavg
ok 123 - get_memory
ok 124 - get_osfhandle_valid_handle
ok 125 - getaddrinfo_basic
ok 126 - getaddrinfo_basic_sync
ok 127 - getaddrinfo_concurrent
ok 128 - gethostname
ok 129 - getsockname_tcp
ok 130 - getsockname_udp
ok 131 - getters_setters
ok 132 - gettimeofday
ok 133 - handle_fileno
ok 134 - handle_type_name
ok 135 - has_ref
ok 136 - homedir
ok 137 - hrtime
ok 138 - idle_check
ok 139 - idle_ref
ok 140 - idle_starvation
ok 141 - idna_toascii
ok 142 - iouring_pollhup
ok 143 - ip4_addr
ok 144 - ip6_addr_link_local
ok 145 - ip6_pton
ok 146 - ip6_sin6_len
ok 147 - ip_name
ok 148 - ipc_heavy_traffic_deadlock_bug
ok 149 - ipc_listen_after_write
ok 150 - ipc_listen_before_write
ok 151 - ipc_send_recv_pipe
ok 152 - ipc_send_recv_pipe_inprocess
ok 153 - ipc_send_recv_tcp
ok 154 - ipc_send_recv_tcp_inprocess
ok 155 - ipc_send_zero
ok 156 - ipc_tcp_connection
ok 157 - kill
ok 158 - kill_invalid_signum
ok 159 - loop_alive
ok 160 - loop_backend_timeout
ok 161 - loop_close
ok 162 - loop_configure
ok 163 - loop_handles
ok 164 - loop_instant_close
ok 165 - loop_stop
ok 166 - loop_stop_before_run
ok 167 - loop_update_time
ok 168 - metrics_idle_time
ok 169 - metrics_idle_time_thread
ok 170 - metrics_idle_time_zero
ok 171 - metrics_info_check
ok 172 - multiple_listen
ok 173 - not_readable_nor_writable_on_read_error
ok 174 - not_writable_after_shutdown
ok 175 - open_osfhandle_valid_handle
ok 176 - pipe_bind_error_addrinuse
ok 177 - pipe_bind_error_addrnotavail
ok 178 - pipe_bind_error_inval
ok 179 - pipe_bind_or_listen_error_after_close
ok 180 - pipe_close_stdout_read_stdin
ok 181 - pipe_connect_bad_name
ok 182 - pipe_connect_close_multiple
ok 183 - pipe_connect_multiple
ok 184 - pipe_connect_on_prepare
ok 185 - pipe_connect_to_file
ok 186 - pipe_getsockname
ok 187 - pipe_getsockname_abstract
ok 188 - pipe_getsockname_autobind
ok 189 - pipe_getsockname_blocking
ok 190 - pipe_listen_without_bind
ok 191 - pipe_overlong_path
ok 192 - pipe_pending_instances
ok 193 - pipe_ping_pong
ok 194 - pipe_ping_pong_vec
ok 195 - pipe_ref
ok 196 - pipe_ref2
ok 197 - pipe_ref3
ok 198 - pipe_ref4
ok 199 - pipe_sendmsg
ok 200 - pipe_server_close
ok 201 - pipe_set_chmod
ok 202 - pipe_set_non_blocking
ok 203 - poll_bad_fdtype
ok 204 - poll_close
ok 205 - poll_close_doesnt_corrupt_stack # SKIP Test only relevant on Windows
ok 206 - poll_closesocket # SKIP Test only relevant on Windows
ok 207 - poll_duplex
ok 208 - poll_multiple_handles
ok 209 - poll_nested_epoll
ok 210 - poll_oob
ok 211 - poll_unidirectional
ok 212 - prepare_ref
ok 213 - process_priority
ok 214 - process_ref
ok 215 - process_title
ok 216 - process_title_big_argv
ok 217 - process_title_threadsafe
ok 218 - queue_foreach_delete
ok 219 - random_async
ok 220 - random_sync
ok 221 - readable_on_eof
ok 222 - ref
ok 223 - req_type_name
ok 224 - run_nowait
ok 225 - run_once
ok 226 - semaphore_1
ok 227 - semaphore_2
ok 228 - semaphore_3
ok 229 - shutdown_close_pipe
ok 230 - shutdown_close_tcp
ok 231 - shutdown_eof
ok 232 - shutdown_simultaneous
ok 233 - shutdown_twice
ok 234 - signal_close_loop_alive
ok 235 - signal_multiple_loops
ok 236 - signal_pending_on_close
ok 237 - socket_buffer_size
ok 238 - spawn_and_kill
ok 239 - spawn_and_kill_with_std
ok 240 - spawn_and_ping
ok 241 - spawn_auto_unref
ok 242 - spawn_closed_process_io
ok 243 - spawn_detached
ok 244 - spawn_empty_env
ok 245 - spawn_exercise_sigchld_issue
ok 246 - spawn_exit_code
ok 247 - spawn_fails
ok 248 - spawn_fails_check_for_waitpid_cleanup
ok 249 - spawn_fs_open
ok 250 - spawn_ignored_stdio
ok 251 - spawn_inherit_streams
ok 252 - spawn_preserve_env
ok 253 - spawn_quoted_path # SKIP Test for Windows
ok 254 - spawn_reads_child_path
ok 255 - spawn_relative_path
ok 256 - spawn_same_stdout_stderr
ok 257 - spawn_setuid_setgid # SKIP It should be run as root user
ok 258 - spawn_stdin
ok 259 - spawn_stdio_greater_than_3
ok 260 - spawn_stdout
ok 261 - spawn_stdout_and_stderr_to_file
ok 262 - spawn_stdout_and_stderr_to_file2
ok 263 - spawn_stdout_and_stderr_to_file_swap
ok 264 - spawn_stdout_to_file
ok 265 - spawn_tcp_server
ok 266 - stdio_emulate_iocp
ok 267 - stdio_over_pipes
ok 268 - strscpy
ok 269 - strtok
ok 270 - sys_error
ok 271 - tcp6_local_connect_timeout
ok 272 - tcp6_ping_pong
ok 273 - tcp6_ping_pong_vec
ok 274 - tcp_alloc_cb_fail
ok 275 - tcp_bind6_error_addrnotavail
ok 276 - tcp_bind6_error_fault
ok 277 - tcp_bind6_error_inval
ok 278 - tcp_bind6_localhost_ok
ok 279 - tcp_bind_error_addrinuse_connect
ok 280 - tcp_bind_error_addrnotavail_1
ok 281 - tcp_bind_error_addrnotavail_2
ok 282 - tcp_bind_error_fault
ok 283 - tcp_bind_error_inval
ok 284 - tcp_bind_invalid_flags
ok 285 - tcp_bind_localhost_ok
ok 286 - tcp_bind_or_listen_error_after_close
ok 287 - tcp_bind_writable_flags
ok 288 - tcp_close
ok 289 - tcp_close_accept
ok 290 - tcp_close_after_read_timeout
ok 291 - tcp_close_reset_accepted
ok 292 - tcp_close_reset_accepted_after_shutdown
ok 293 - tcp_close_reset_accepted_after_socket_shutdown
ok 294 - tcp_close_reset_client
ok 295 - tcp_close_reset_client_after_shutdown
ok 296 - tcp_close_while_connecting # SKIP Network unreachable.
ok 297 - tcp_connect6_error_fault
ok 298 - tcp_connect_error_after_write
ok 299 - tcp_connect_error_fault
ok 300 - tcp_create_early
ok 301 - tcp_create_early_accept
ok 302 - tcp_create_early_bad_bind
ok 303 - tcp_create_early_bad_domain
ok 304 - tcp_flags
ok 305 - tcp_listen_without_bind
ok 306 - tcp_local_connect_timeout
ok 307 - tcp_oob
ok 308 - tcp_open
ok 309 - tcp_open_bound
ok 310 - tcp_open_connected
ok 311 - tcp_open_twice
ok 312 - tcp_ping_pong
ok 313 - tcp_ping_pong_vec
ok 314 - tcp_read_stop
ok 315 - tcp_read_stop_start
ok 316 - tcp_ref
ok 317 - tcp_ref2
ok 318 - tcp_ref2b
ok 319 - tcp_ref3
ok 320 - tcp_ref4
ok 321 - tcp_reuseport
ok 322 - tcp_rst
ok 323 - tcp_shutdown_after_write
ok 324 - tcp_try_write
ok 325 - tcp_try_write_error
ok 326 - tcp_unexpected_read
ok 327 - tcp_write_after_connect
ok 328 - tcp_write_fail
ok 329 - tcp_write_in_a_row
ok 330 - tcp_write_queue_order
ok 331 - tcp_write_ready
ok 332 - tcp_write_to_half_open_connection
ok 333 - test_macros
not ok 334 - thread_affinity
# exit code 134
# Output from process `thread_affinity`:
# Assertion failed in test/test-thread-affinity.c on line 73: cpumask[0] && "test must be run with cpu 0 affinity"
ok 335 - thread_create
ok 336 - thread_equal
ok 337 - thread_local_storage
ok 338 - thread_mutex
ok 339 - thread_mutex_recursive
ok 340 - thread_rwlock
ok 341 - thread_rwlock_trylock
ok 342 - thread_stack_size
ok 343 - thread_stack_size_explicit
ok 344 - threadpool_cancel_fs
ok 345 - threadpool_cancel_getaddrinfo
ok 346 - threadpool_cancel_getnameinfo
ok 347 - threadpool_cancel_random
ok 348 - threadpool_cancel_single
ok 349 - threadpool_cancel_when_busy
ok 350 - threadpool_cancel_work
ok 351 - threadpool_queue_work_einval
ok 352 - threadpool_queue_work_simple
ok 353 - timer
ok 354 - timer_again
ok 355 - timer_early_check
ok 356 - timer_from_check
ok 357 - timer_huge_repeat
ok 358 - timer_huge_timeout
ok 359 - timer_init
ok 360 - timer_is_closing
ok 361 - timer_no_double_call_nowait
ok 362 - timer_no_double_call_once
ok 363 - timer_no_run_on_unref
ok 364 - timer_null_callback
ok 365 - timer_order
ok 366 - timer_ref
ok 367 - timer_ref2
ok 368 - timer_run_once
ok 369 - timer_start_twice
ok 370 - timer_zero_timeout
ok 371 - tmpdir
ok 372 - tty # SKIP Cannot open /dev/tty as read-only: No such device or address
ok 373 - tty_file
ok 374 - tty_pty
ok 375 - udp_alloc_cb_fail
ok 376 - udp_bind
ok 377 - udp_bind_reuseaddr
ok 378 - udp_connect
ok 379 - udp_connect6
ok 380 - udp_create_early
ok 381 - udp_create_early_bad_bind
ok 382 - udp_create_early_bad_domain
ok 383 - udp_dgram_too_big
ok 384 - udp_dual_stack
ok 385 - udp_ipv6_only
ok 386 - udp_mmsg
ok 387 - udp_multicast_interface
ok 388 - udp_multicast_interface6
ok 389 - udp_multicast_ttl
ok 390 - udp_no_autobind
ok 391 - udp_open
ok 392 - udp_open_bound
ok 393 - udp_open_connect
ok 394 - udp_open_twice
ok 395 - udp_options
ok 396 - udp_options6
ok 397 - udp_recv_in_a_row
ok 398 - udp_ref
ok 399 - udp_ref2
ok 400 - udp_ref3
ok 401 - udp_reuseport
ok 402 - udp_send_and_recv
ok 403 - udp_send_immediate
ok 404 - udp_send_unix
ok 405 - udp_send_unreachable
ok 406 - udp_sendmmsg_error
ok 407 - udp_try_send
ok 408 - uname
ok 409 - unref_in_prepare_cb
ok 410 - utf8_decode1
ok 411 - utf8_decode1_overrun
ok 412 - walk_handles
ok 413 - watcher_cross_stop
ok 414 - we_get_signal
ok 415 - we_get_signal_one_shot
ok 416 - we_get_signals
ok 417 - we_get_signals_mixed
FAIL: test/run-tests
======================================================
1 of 1 test failed
Please report to https://github.com/libuv/libuv/issues
======================================================
make[1]: *** [Makefile:5709: check-TESTS] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:5974: check-am] Error 2
```

### Additional context

Note that I ran into this issue when changing `gcc.arch = "x86-64-v3"`, and there is another issue with libuv and changing gcc.arch here: #338114, but I have patched that. I believe this came up when changing my gcc.arch only because that has caused me not to fetch a pre-built libuv. My flake compiles libuv fine when not run in my gitea action.

Would it be best to just disable this test? I feel that nix should be able to build packages with any thread affinity.




### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.63, NixOS, 25.05 (Warbler), 25.05.20250116.5df4362`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.25.3`
 - channels(root): `"nixos-25.05beta708622.5e4fbfb6b3de"`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`

### Notify maintainers

_No response_

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/libuv/default.nix
+++ b/pkgs/development/libraries/libuv/default.nix
@@ -52,6 +52,7 @@ stdenv.mkDerivation (finalAttrs: {
           "getaddrinfo_fail"
           "getaddrinfo_fail_sync"
           "tcp_connect6_link_local"
+          "thread_affinity" # else "test must be run with cpu 0 affinity" when affinity is set
           "threadpool_multiple_event_loops" # times out on slow machines
           "get_passwd" # passed on NixOS but failed on other Linuxes
           "tcp_writealot"

